app/src/org/commcare/utils/CommCareUtil.java

```diff
@@ -82,8 +82,7 @@ public static int countPreds(TreeReference reference) {
     }
 
     public static void triggerLogSubmission(Context c) {
-        SharedPreferences settings = CommCareApplication.instance().getCurrentApp().getAppPreferences();
-        String url = settings.getString(CommCareServerPreferences.PREFS_SUBMISSION_URL_KEY, null);
+        String url = LogSubmissionTask.getSubmissionUrl(CommCareApplication.instance().getCurrentApp().getAppPreferences());
```

Comment:
I feel like it's an antipattern to increase the number of places in the code where a static singleton is requested from a static context, so I felt like it was wrong to do so. If you feel strongly will move it.